ethereum_bridge/src/protocol/transactions/ethereum_events/events.rs

```diff
@@ -71,20 +72,78 @@ where
         receiver,
     } in transfers
     {
-        if asset != &wrapped_native_erc20 {
-            let mut changed =
-                mint_wrapped_erc20s(storage, asset, receiver, amount)?;
-            changed_keys.append(&mut changed)
+        let mut changed = if asset != &wrapped_native_erc20 {
+            mint_wrapped_erc20s(storage, asset, receiver, amount)?
         } else {
-            tracing::warn!(
-                "Redemption of the wrapped native token is not yet supported \
-                 - (receiver - {receiver}, amount - {amount})"
-            )
-        }
+            redeem_native_token(storage, receiver, amount)?
+        };
+        changed_keys.append(&mut changed)
     }
     Ok(changed_keys)
 }
 
+/// Redeems `amount` of the native token for `receiver` from escrow.
+fn redeem_native_token<D, H>(
+    storage: &mut Storage<D, H>,
+    receiver: &Address,
+    amount: &token::Amount,
+) -> Result<BTreeSet<Key>>
+where
+    D: 'static + DB + for<'iter> DBIter<'iter> + Sync,
+    H: 'static + StorageHasher + Sync,
+{
+    let bridge_pool_native_token_balance_key =
```

Comment:
Escrowed NAM is held under th eth bridge vp, not the bridge pool.

Reply:
Ah, have updated